lib/ftpWrapper.js

```diff
@@ -0,0 +1,66 @@
+var ftp = require('ftp');
+
+var clients = {};
+
+function connect(urlObj, callback, errback){
+
+  if(clients[urlObj.hostname]) return process.nextTick(function(){
+    return callback(null, clients[urlObj.hostname]);
+  });
+
+  var client = new ftp();
+  var beforeCallback = 1;
+  clients[urlObj.hostname] = client;
+
+  client.on('ready', function(){
+    beforeCallback = 0;
+    return callback(null, client);
+  });
+
+  client.on('error', function(err){
+    delete clients[urlObj.hostname];
+    if(errback) errback(err);
+    if(beforeCallback) callback(err);
+  });
+
+  client.connect({host: urlObj.hostname, port: urlObj.port || '21'});
+}
+
+
+function request(urlObj, callback){
+
+  var client = clients[urlObj.hostname];
+  if(!client) return callback(new Error('Ftp client not connected'));
```

Comment:
Probably worth adding the `hostname` and any other useful info to the error message.  It's hard for me to tell if this is already logged somewhere else.  If not, something like "FTP client for ftp://ftp.somestate.gov not connected" provides a bit more context.